### PR TITLE
🏗️ Add `inspect()`

### PR DESCRIPTION
**Pull request by Zethson**
Solves #354 #353 #349

- [x] Play with BiontyField and implement it
- [x] Rename Entity to Bionty
- [x] Fix tests
- [x] Implement inspect
- [x] Add inspect tests
- [x] Discuss & review intermediate with Sunny
- [x] Adapt docs

### CODE REVIEW

File: bionty/_display_versions.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def display_available_versions(
 
     table.add_column("Ontology", justify="right", style="cyan", no_wrap=True)
     table.add_column("URL", justify="right", style="cyan", no_wrap=True)
-    table.add_column("Bionty Entity", justify="right", style="cyan", no_wrap=True)
+    table.add_column("Bionty Bionty", justify="right", style="cyan", no_wrap=True)
```
Comment (sunnyosun) [inline bionty/_display_versions.py]: Maybe call this "Bionty class"?

File: bionty/_entity.py [hunk 2]
```diff
@@ -40,7 +38,7 @@ def __init__(
         version: Optional[str] = None,
         species: Optional[str] = None,
         *,
-        reference_id: Optional[str] = None,
+        reference_id: Optional[Union[BiontyField, str]] = None,
```
Comment (sunnyosun) [inline bionty/_entity.py:41]: I'd just remove this `reference_id` here and have `.df(reference_id="...")`, what do you think?

File: bionty/_normalize.py [hunk 3]
```diff
@@ -32,10 +32,10 @@ class NormalizeColumns:
 
     @staticmethod
     def gene(df: pd.DataFrame, species=None):
-        """Column names of gene Entity."""
+        """Column names of gene Bionty."""
```
Comment (sunnyosun) [inline bionty/_normalize.py]: """Column names of Gene().df."""

File: bionty/_normalize.py [hunk 4]
```diff
@@ -32,10 +32,10 @@ class NormalizeColumns:
...
         df.rename(columns=GENE_COLUMNS, inplace=True)
 
     @staticmethod
     def protein(df: pd.DataFrame, species=None):
-        """Column names of protein Entity."""
+        """Column names of protein Bionty."""
```
Comment (sunnyosun) [inline bionty/_normalize.py]: """Column names of Protein().df."""

File: bionty/dev/_fix_index.py [hunk 5]
```diff
@@ -39,7 +39,7 @@ def get_compliant_index_from_column(
     Args:
         df: DataFrame with an index different from reference ID, but with a
             column that has mappable information, like an ensemble_id.
-        ref_df: Reference Entity in Bionty.
+        ref_df: Reference Bionty in Bionty.
```
Comment (sunnyosun) [inline bionty/dev/_fix_index.py]: Reference table in Bionty.

File: bionty/dev/_handle_versions.py [hunk 6]
```diff
@@ -149,7 +149,7 @@ def _get_missing_defaults(
                   Defaults to "current".
 
     Returns:
-        A list of tuples in the form of [(Entity, source, version)] that
+        A list of tuples in the form of [(Bionty, source, version)] that
```
Comment (sunnyosun) [inline bionty/dev/_handle_versions.py]: class, source, version?

**Review by sunnyosun — APPROVED** [review]

**Review by sunnyosun — APPROVED** [review]

File: bionty/_display_versions.py (hunk 1 above)
Comment (Zethson) [reply bionty/_display_versions.py]: That's a funny one. Happened due to the automatic renaming :)

File: bionty/_entity.py (hunk 2 above)
Comment (Zethson) [reply bionty/_entity.py:41]: See Slack - not a fan of this if I don't miss anything.

**Pull request merged**